lib/UmdMainTemplatePlugin.js

```diff
@@ -124,16 +134,16 @@ class UmdMainTemplatePlugin {
 				) +
 				"	else if(typeof define === 'function' && define.amd)\n" +
 				(requiredExternals.length > 0 ?
-					(this.name && this.namedDefine === true ?
-						"		define(" + libraryName(this.name) + ", " + externalsDepsArray(requiredExternals) + ", " + amdFactory + ");\n" :
+					(this.names.amd && this.namedDefine === true ?
+						"		define(" + libraryName(this.names.amd) + ", " + externalsDepsArray(requiredExternals) + ", " + amdFactory + ");\n" :
 						"		define(" + externalsDepsArray(requiredExternals) + ", " + amdFactory + ");\n"
 					) :
-					(this.name && this.namedDefine === true ?
-						"		define(" + libraryName(this.name) + ", [], " + amdFactory + ");\n" :
+					(this.names.amd && this.namedDefine === true ?
+						"		define(" + libraryName(this.names.amd) + ", [], " + amdFactory + ");\n" :
```

Comment:
this line is not tested, if you want to you can add a test, but it's not required.